src/BCFKeyValueData.cc

```diff
@@ -874,6 +880,24 @@ static Status validate_bcf(BCFBucketRange& rangeHelper, const bcf_hdr_t *hdr,
                                std::to_string(prev_pos) + " >= " + range(bcf).str());
     }
 
+    // verify record does not go over the length of the contig
+    if (bcf->pos + bcf->rlen > contigLenVec[bcf->rid]) {
+        return Status::Invalid("gVCF record is longer than contig",
+                               range(bcf).str() + " " + to_string(contigLenVec[bcf->rid]));
+    }
+
+    // DNA regular expression for REF and ALT alleles
+    std::regex e ("[ACGTN]+");
+    for (int i=0; i < bcf->n_allele; i++) {
+        const string allele_i(bcf->d.allele[i]);
+        if (allele_i == "<NON_REF>" ||
```

Comment:
Let's accept "<.*>" (or just check if the first character is `<` and the last character is `>`). Unfortunately different variant callers use different symbolic alleles in the gVCF records.